buildcontext/gitlookup.go

```diff
@@ -177,6 +177,16 @@ var errUnsupportedHash = errors.New("unsupported keyscan hash")
 var errInvalidScan = errors.New("invalid keyscan")
 var errKeyScanNoMatch = errors.New("keyscan does not match")
 
+func hasPort(s string) bool {
```

Comment:
There's nothing in the net package to check this?

Reply:
Maybe SplitHostPort returns a specific error if the port is missing?

Reply:
`SplitHostPort` isn't great; I tried that originally. 

https://cs.opensource.google/go/go/+/refs/tags/go1.17.5:src/net/ipsock.go;l=167 sets `tooManyColons = "too many colons in address"` then creates an error on the fly `return addrErr(hostport, tooManyColons)`. My only other idea was to try `strings.Contains(err.Error, "too many colons in address")`, but that feels more error-prone as that error string might change.